index.bs

```diff
@@ -439,29 +449,33 @@ interface FontManager {
 };
 
 dictionary QueryOptions {
-  sequence<DOMString> select = [];
+  sequence<DOMString> postscriptNames;
 };
 </xmp>
 
 <div algorithm>
 The <dfn method for=FontManager>query(|options|)</dfn> method steps are:
 
 1. Let |promise| be [=/a new promise=].
-1. If [=/this=]’s [=relevant settings object=]'s [=origin=] is an [=/opaque origin=], then [=/reject=] |promise| with a {{TypeError}}, and return |promise|.
+1. Let |descriptor| be a {{PermissionDescriptor}} with its {{PermissionDescriptor/name}} set to {{PermissionName/"font-access"}}.
+1. If [=/this=]’s [=relevant settings object=]'s [=origin=] is an [=/opaque origin=], then [=/reject=] |promise| with a "{{SecurityError}}" {{DOMException}}, and return |promise|.
 1. If [=/this=]’s [=relevant global object=]'s [=/associated Document=] is not [=/allowed to use=] the [=/policy-controlled feature=] named {{PermissionPolicy/"local-fonts"}}, then [=/reject=] |promise| with a "{{SecurityError}}" {{DOMException}}, and return |promise|.
 1. If [=/this=]’s [=relevant global object=] does not have [=/transient activation=], then [=/reject=] |promise| with a "{{SecurityError}}" {{DOMException}}, and return |promise|.
 1. Otherwise, run these steps [=in parallel=]:
-    1. Let |permissionState| be [=request permission to use=] {{PermissionName/"font-access"}}.
-    1. If |permissionState| is {{PermissionState/"denied"}}, then [=/reject=] |promise| with a "{{NotAllowedError}}" {{DOMException}}, and abort these steps.
-    1. Let |fonts| be a [=/list=] of all local fonts on the system.
-    1. Let |select| be |options|' {{QueryOptions/"select"}} member.
-    1. Let |result| be an new [=/list=].
-    1. [=list/For each=] font |font| in |fonts|, run these steps:
+    1. Let |system fonts| be a [=/list=] of all local fonts on the system.
+    1. Let |selectable fonts| be a new [=/list=].
+    1. [=list/For each=] font |font| in |system fonts|, run these steps:
         1. Let |representation| be a [=/font representation=] for |font|.
-        1. Let |postscriptName| be |representation|'s [=font representation/name string=] 6 for \``en`\`.
-        1. If |select| is empty, or if |select| [=list/contains=] |postscriptName|, then [=list/append=] a new {{FontMetadata}} instance associated with |representation| to |result|.
-    1. Sort |list| in [=list/sort in ascending order|ascending order=] by using {{FontMetadata/postscriptName}} as the sort key and store the result as |list|.
-    1. [=/Resolve=] |promise| with |list|.
+        1. Set |representation|'s [=font representation/data bytes=] to a SFNT [[!SFNT]] serialization of the representation.
+        1. Let |postscriptName| be |representation|'s [=font representation/PostScript name=].
+        1. If |postscriptName| is not valid according to the [[!OPENTYPE]] <a href="https://docs.microsoft.com/en-us/typography/opentype/spec/name">naming table</a> specification, then [=iteration/continue=].
```

Comment:
Do you want to also give UAs a choice to eliminate fonts from the list on other grounds, maybe privacy? E.g. "If the UA determines that the user should never expose a particular font to the web, it may [=iteration/continue=]."